mmv1/third_party/terraform/fwprovider/framework_provider_user_project_override_test.go.erb

```diff
@@ -0,0 +1,350 @@
+<% autogen_exception -%>
+package fwprovider_test
+
+import (
+	"fmt"
+<% unless version == 'ga' -%>
+	"regexp"
+<% end -%>
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+<% unless version == 'ga' -%>
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+<% end -%>
+)
+
+// TestAccFwProvider_user_project_override is a series of acc tests asserting how the plugin-framework provider handles credentials arguments
+// It is PF specific because the HCL used uses a PF-implemented data source
+// It is a counterpart to TestAccSdkProvider_user_project_override
+func TestAccFwProvider_user_project_override(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		// Configuring the provider using inputs
+		"config takes precedence over environment variables":                                testAccFwProvider_user_project_override_configPrecedenceOverEnvironmentVariables,
+		"when user_project_override is unset in the config, environment variables are used": testAccFwProvider_user_project_override_precedenceOrderEnvironmentVariables,
+
+		// Schema-level validation
+		"when user_project_override is set in the config the value can be a boolean (true/false) or a string (true/false/1/0)": testAccFwProvider_user_project_override_booleansInConfigOnly,
+		"when user_project_override is set via environment variables any of these values can be used: true/false/1/0":          testAccFwProvider_user_project_override_envStringsAccepted,
+
+		// Usage
+<% unless version == 'ga' -%>
+		"user_project_override controls which project is used for quota and billing purposes": testAccFwProvider_UserProjectOverride,
+		// We cannot currently implement this test case in a PF-specific way: "user_project_override works for resources that don't take a project id as an argument directly"
```

Comment:
Currently the only things implemented with the plugin framework are:

- google_client_config
- google_client_openid_userinfo
- google_firebase_web_app_config (beta)
- google_firebase_android_app_config (beta)
- google_firebase_apple_app_config (beta)

The first two don't interact with managed resources so aren't relevant to user_project_override, so that leaves only the Beta-only Firebase stuff. This is why the test needs to be templated in a GA/Beta way. In future, when more things are migrated to the PF, this test can be simplified.

Reply:
Also, these test cases mimic what was originally done in TestAccProviderUserProjectOverride and TestAccProviderIndirectUserProjectOverride